DemoApp/Shared/Token+Development.swift

```diff
@@ -0,0 +1,79 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+extension StreamChatWrapper {
+    func refreshingTokenProvider(initialToken: Token, tokenDurationInMinutes: Double) -> TokenProvider {
+        { completion in
+            // Simulate API call delay
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.3) {
+                let token: Token
+                #if GENERATE_JWT
+                let timeInterval = TimeInterval(tokenDurationInMinutes * 60)
+                let generatedToken = _generateUserToken(
+                    secret: appSecret,
+                    userID: initialToken.userId,
+                    expirationDate: Date().addingTimeInterval(timeInterval)
+                )
+                if generatedToken == nil {
+                    print("Unable to generate token. Falling back to initialToken")
+                }
+                token = generatedToken ?? initialToken
+                #else
+                token = initialToken
+                #endif
+                completion(.success(token))
+            }
+        }
+    }
+}
+
+#if GENERATE_JWT
+
+import CryptoKit
+import Foundation
+import StreamChat
+
+let appSecret = ""
```

Comment:
How's the secret going to be set? Manually or would there be another way?

Reply:
As of now, manually. We don't want anyone to be using this